proto/wtools/abase/l0/l1.test/Array.test.s

```diff
@@ -1665,36 +1665,240 @@ function from( test )
 
 function as( test )
 {
-  test.case = 'an empty array';
-  var got = _.array.as( [] );
-  var expected = [];
-  test.identical( got, expected );
-
-  test.case = 'null';
-  var got = _.array.as( null );
-  var expected = [];
-  test.identical( got, expected );
-
-  test.case = 'array contains an object';
-  var got = _.array.as( { a : 1, b : 2 } );
-  var expected = [ { a : 1, b : 2 } ];
-  test.identical( got, expected );
+  asTemplate( { method : 'as' } );
+  asTemplate( { method : 'asTest' } );
 
-  test.case = 'array contains boolean';
-  var got = _.array.as( true );
-  var expected = [ true ];
-  test.identical( got, expected );
-
-  /**/
-
-  if( !Config.debug )
-  return;
-
-  test.case = 'nothing';
-  test.shouldThrowErrorSync( () => _.array.as() );
-
-  test.case = 'undefined';
-  test.shouldThrowErrorSync( () => _.array.as( undefined ) );
+  function asTemplate( env )
+  {
+    test.case = `${__.entity.exportStringSolo( env )}, null`;
+    var got = _.array[ env.method ]( null );
+    var expected = [];
+    test.identical( got, expected );
+
+    /* */
+
+    test.case = `${__.entity.exportStringSolo( env )}, an empty array`;
+    var src = [];
+    var got = _.array[ env.method ]( src );
+    var expected = src;
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an array`;
+    var src = [ 1, 2, 3 ]
+    var got = _.array[ env.method ]( src );
+    var expected = src;
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an Array Prototype`;
+    var src = Array.prototype;
+    var got = _.array[ env.method ]( src );
+    if( env.method === 'asTest')
+    var expected = [ ... src ];
+    if( env.method === 'as')
+    var expected = src;
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an array having generator function as it's Symbol.iterator`;
+    var src = [];
+    src[ Symbol.iterator ] = function* ()
+    {
+      yield 1;
+      yield 2;
+      yield 3;
+    };
+    var got =_.array[ env.method ]( src );
+    if( env.method === 'asTest')
+    var expected = [ ... src ];
+    if( env.method === 'as')
```

Comment:
- src[0] is undefined, [ ... src ][0] is 1
- [ ... src ] as the expected output seems more appropriate than src as the expected output.